components/forward_post_modal/forward_post_modal.tsx

```diff
@@ -0,0 +1,434 @@
+                selectionStart,
+                selectionEnd,
+                message: value,
+            });
+        } else if (ctrlKeyCombo && Utils.isKeyPressed(e, KeyCodes.B)) {
+            applyMarkdownMode({
+                markdownMode: 'bold',
+                selectionStart,
+                selectionEnd,
+                message: value,
+            });
+        } else if (ctrlKeyCombo && Utils.isKeyPressed(e, KeyCodes.I)) {
+            applyMarkdownMode({
+                markdownMode: 'italic',
+                selectionStart,
+                selectionEnd,
+                message: value,
+            });
+        } else if (ctrlShiftCombo && Utils.isKeyPressed(e, KeyCodes.X)) {
+            applyMarkdownMode({
+                markdownMode: 'strike',
+                selectionStart,
+                selectionEnd,
+                message: value,
+            });
+        } else if (ctrlShiftCombo && Utils.isKeyPressed(e, KeyCodes.E)) {
+            e.stopPropagation();
+            e.preventDefault();
+        } else if (ctrlEnterKeyCombo && canForwardPost) {
+            handleSubmit();
+        }
+    };
+
+    const handleSelect = (e: React.SyntheticEvent<Element, Event>) => {
+        Utils.adjustSelection(textboxRef.current?.getInputBox(), e as React.KeyboardEvent<HTMLInputElement>);
+    };
+
+    const handlePostError = (postError: React.ReactNode) => setPostError(postError);
+
+    // we don't care in this textbox about it
+    const handleHeightChange = Utils.noop;
+
+    // we don't care in this textbox about it
+    const handleMouseUpKeyUp = Utils.noop;
+
+    // this does not make any sense in this modal, so we add noop here
+    const emitTypingEvent = Utils.noop;
+
+    // we do not allow sending the forwarding when hitting enter
+    const postMsgKeyPress = Utils.noop;
+
+    // we do not care about the blur event
+    const handleBlur = Utils.noop;
+
+    // since the original post has a click handler specified we should prevent any action here
+    const preventActionOnPreview = (e: React.MouseEvent) => {
+        e.preventDefault();
+        e.stopPropagation();
+    };
+
+    const createMessage = formatMessage({id: 'forward_post_modal.comment.placeholder', defaultMessage: 'Add a comment (optional)'});
+    const messagePreviewTitle = formatMessage({id: 'forward_post_modal.preview.title', defaultMessage: 'Message Preview'});
+    const previewMetaData: PostPreviewMetadata = {
+        post,
+        post_id: post.id,
+        team_name: currentTeam.name,
+        channel_display_name: currentChannel.display_name,
+        channel_type: currentChannel.type,
+        channel_id: currentChannel.id,
+    };
+
+    let notification;
+    if (isPrivateConversation) {
+        let notificationText;
+        if (currentChannel.type === General.PRIVATE_CHANNEL) {
+            const channel = `~${currentChannel.display_name}`;
+            notificationText = (
+                <FormattedMarkdownMessage
```

Comment:
Hmm, that's a good question. Even without the strong tags, that actually wouldn't work since the `and` wouldn't get translated. I think we might be able to do something like using two different translation strings, one for a single user and one for multiple sort of like we do it for the [`CombinedSystemMessge`'s "userA and userB have joined the channel"](https://github.com/mattermost/mattermost-webapp/blob/master/components/post_view/combined_system_message/combined_system_message.tsx#L24). Then, instead of passing in a string for `participants`, then we could pass in a React object that looks like
```js
<FormattedMessage
    defaultMessage='Forwarded to {participants} and {lastParticipant}.'
    values={{
        participants: (
            <>
                <strong>{'user1'}</strong>
                {', '}
                <strong>{'user2'}</strong>
            <>
        ),
        lastParticipant: <strong>{'user3'}</strong>
    }}
/>
```

Reply:
Actually, I just discovered that [FormattedList](https://formatjs.io/docs/react-intl/components/#formattedlist) exists. That'd make this much easier
```js
const participants = allParticipants.map((participant) => <strong>{participant}</strong>);

<FormattedMessage
    defaultMessage='Forwarded to {participants}.'
    values={{
        participants: <FormattedList values={participants}/>,
    }}
/>
``` 
Hopefully there's no languages that translate the "Forwarded to" part differently when participants is singular or plural. There's ways around that though

Reply:
nice find!